parcels/kernel.py

```diff
@@ -275,8 +275,11 @@ def execute_python(self, pset, endtime, dt):
                 for var in ptype.variables:
                     p_var_back[var.name] = getattr(p, var.name)
                 try:
-                    p.dt = sign_dt * dt_pos
+                    pdt = sign_dt * dt_pos
+                    p.dt = pdt
```

Comment:
These two lines are somewhat unclear. Would it help to rename `pdt` to something clearer, e.g.
```
p.dt =  sign_dt * dt_pos
pdt_prekernels = p.dt
```
and then check `... not np.isclose(p.dt, pdt_prekernels)` below?